modules/Core/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryGrid.java

```diff
@@ -55,6 +55,23 @@
     private List<InventoryCell> cells = Lists.newArrayList();
     private Binding<EntityRef> targetEntity = new DefaultBinding<>(EntityRef.NULL);
 
+    private InteractionListener interactionListener = new BaseInteractionListener() {
+        @Override
+        public boolean onMouseClick(NUIMouseClickEvent event) {
+            MouseInput mouseButton = event.getMouseButton();
+            if (mouseButton == MouseInput.MOUSE_LEFT) {
+                LocalPlayer localPlayer = CoreRegistry.get(LocalPlayer.class);
+                EntityRef movingItem = CoreRegistry.get(LocalPlayer.class).getCharacterEntity().getComponent(CharacterComponent.class).movingItem;
```

Comment:
I tried to add ```@RegisterSystem``` to ```InventoryGrid``` and then added a field with:
```
@In
private LocalPlayer localPlayer;
```
But it doesn't work. The variable is null at runtime.
Also, I tried to implement ```ComponentSystem``` in InventoryGrid but that didn't work too.
.
I did these things because all the classes with ```@In``` variable had atleast one of these 2 features.
How do I do that?